### ISSUE TITLE
Fan won't switch off when under the thresholds set in the configuration

### ISSUE THREAD

**Issue by rhavish** [adamoutler/HassOSArgonOneAddon#7]
I installed this addon today on my RPI4. Installation went really smoothly and I used the HassOS I2C Configurator that was referenced to enable I2C.

Only issue I’m having is actually switching the fan off fully.

My config is as follows:

```
CorF: C
LowRange: 55
MediumRange:60
HighRange:65
QuietProfile: Ticked
Create a Fan Speed entity in Home Assistant: Ticked
```

My Pi is consistently around the 34/35c temperature. The log is below.

```
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: ArgonOne Active Cooling
 Actively keeping your Argon One cool.
-----------------------------------------------------------
 Add-on version: 20
 You are running the latest version of this add-on.
 System: Home Assistant OS 5.13  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2021.4.5
 Home Assistant Supervisor: 2021.04.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
Detecting Layout of i2c, we expect to see "1a" here.
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- 1a -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
Settings initialized. Argon One Detected. Beginning monitor..
Current Temperature 38 °C
Level 1 - Fan 0% (OFF)
Current Temperature 36 °C
```

As I find the fan too noisy for the room I’m in and the Pi is not under heavy load, I’d prefer the fan to be off until it hits the ranges defined in the config. Is this possible/the expected behaviour?

Thanks!

**Comment by adamoutler** [comment]
You may have something else interfering.  This addon states it is at 0%.  When at 0% the fan is off.

**Comment by rhavish** [comment]
Ok, makes sense. I can hear the differences in fan speed when I adjust the thresholds, it simply refuses to switch off though.

I intend to troubleshoot this to the best of my ability. Any chance you can offer any starting points please? My first test is to boot up from an SD card with Raspberry Pi OS on it and see if stopping the fan via the original script [here](https://download.argon40.com/argon1.sh) works. Will reply with results.

**Comment by rhavish** [comment]
Fans were still going on Pi OS even when configured to be off. I picked the case up and the fans went off. Turns out I had forgotten to screw the internal board and the SSD enclosure in. I'm still having to play around though as once I got it all screwed in, the fans still wouldn't stop. Dodgy connection somewhere maybe.

I'll close as the issue is not caused by this addon. Wanted to document what I had noticed incase anyone else had a similar issue though.
